### ISSUE TITLE
ES-Lint configuration is invalid -> severity should be one of the following - "error" vs 2

### ISSUE THREAD

**Issue by Ashtonian** [eslint/eslint#5519]
**What version of ESLint are you using?**
2.3
**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
eslint via linter-eslint in atom? with a config file:

``` javascript 
{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "error",
            4
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}
``` 
**What did you do? Please include the actual source code causing the issue.**
It seems to be an issue with any .js file.
**What did you expect to happen?**
I expected it to lint properly 
**What actually happened? Please include the actual, raw output from ESLint.**
Hello! i'm not sure who's this problem is but I recently installed linter-eslint, I went through the eslint init and answered questions. The json object below is the result of that. Atom then seemed to freak out. I changed "error" out for 2 and it seemed to fix it. Please see attached and if this belongs to someone else let me know here to go.

![image](https://cloud.githubusercontent.com/assets/4212452/13613982/f6861df6-e533-11e5-9c4b-4d6b858ca291.png)

Seems to be an issue with the new --init using "error" instead of 2, and the config validator not working. Opened an issue with the linter-eslint folks at https://github.com/AtomLinter/linter-eslint/issues/471 and was told to come here.

**Comment by ilyavolodin** [comment]
It looks like you have global installation of ESLint with version 2.3.0 and local installation with older version. When you ran `init` command, config was generated with new severity levels (`error`, `warn`, `off`), but Atom is using older local version of ESLint, and it fails on the new config file. You should verify version of the local installation of ESLint and upgrade it.

**Comment by tshinnic** [comment]
In my case, I've been reading the latest docs which [happily](http://eslint.org/docs/user-guide/configuring#configuring-rules) say you can use "off", "warn", and "error" and so had been manually updating a .eslintrc file.  But then separately I also had to drop down to 2.2.0 because of the `estraverse-fb` mess (#5476).

So nowhere is it evident (in the docs) that "off" is only *just* *now* supported in 2.3.0 (which is broken and so you might be running a back level which doesn't match the docs no more)

`CHANGELOG.md` sort of mentions this sideways: "Update: Allow string severity in config (fixes #3626)" and now that I look (having found them) the [release notes](http://eslint.org/blog/2016/03/eslint-v2.3.0-released) mention this feature as new to 2.3.0.

Anyway, I wanted to add a note here for other confused people, about this error being generated due to mismatch between docs on website and any need to use a back-level version.

**Comment by ilyavolodin** [comment]
Website have always up to date information for the latest released version. Since 2.3.0 is the latest, and it supports off/error/warn severity levels, that's what website displays.

**Comment by platinumazure** [comment]
It wouldn't hurt to have a note on what version a feature got introduced. We do this with rules, but I'd like to see it in the other doc sections. I think that's something we can try to do better in the future.

**Comment by ilyavolodin** [comment]
We don't really have an organized way to display features (other then rules), so it's hard to figure out a good way to show when the feature was added, but I'm open to suggestions.

**Comment by platinumazure** [comment]
I see it as a final pre-release manual step: the project lead or reviewer
doing the release should do a final commit annotating the docs with "New in
version x.y" for important new things that won't work in earlier versions.
Plus reviewers, when reviewing PRs, should think about whether the feature
is New and they can choose to ask for a note in the docs section of the
commit.

That only works going forward. Looking back, we can make decisions as
people raise issues about whether a version note in the docs would have
been (or still would be) useful. And I can try to go back through the
changelog and file PRs for anything that seems like it could be important
enough to merit an annotation. Of course, the further back the feature was
introduced, the less essential it becomes to add an annotation.

I can open a new issue for further discussion, if needed. I also want to
note that I'm not trying to be critical here- rather, I *really love*
ESLint and think it's a great project, and I just want to give feedback
that I think will help ESLint go from great to unbelievable, by showing an
attention to detail that is simply absent from most other projects.

Thanks for your consideration.

**Comment by nzakas** [comment]
We generally call out important features in our release announcements. This particular change was mentioned here:
http://eslint.org/blog/2016/03/eslint-v2.3.0-released

The changelog is just a list of commits for people who are interested in those; the blog posts are where we put in most of the effort to inform people when important changes are being made. For each release, we go through the commits and add the "Highlights" section into the blog post calling out the important changes. I'm not sure it makes sense to do anything more than that at this point. 

I'm not a fan of littering the documentation with version annotations, though we are planning on making the current stable version more prominent on the website so it's clear what people are looking at. Another option is to keep the documentation separated out for every version so you can always, for instance, look at the v2.2.2 documentation. That discussion should take place on the website repo, though.
